File: AiChallenge/src/main/java/GraphCache.java
/*
 * Copyright (C) Johannes Fähndrich - All Rights Reserved.
 * Unauthorized copying of this file, via any medium is strictly
 * prohibited Proprietary and confidential.
 * Written by Johannes Fähndrich <faehndrich@gmail.com.com>,  2011
 *
 */

import de.dailab.nsm.decomposition.Concept;
import de.tuberlin.spreadalgo.Marker;
import org.jgrapht.Graph;

import java.util.List;
import java.util.Map;

public class GraphCache {
    Question question;
    Graph decGraph ;
    Graph syntaxGraph;
    Graph nerGraph;
    Graph roleGraph;
    Map<String, String> nerMap;
    List<Map<Concept, List<? extends Marker>>> startActivation;

    public GraphCache(Question question,Graph decGraph, Graph syntaxGraph, Graph nerGraph, Graph roleGraph,Map<String, String> nerMapList,List<Map<Concept, List<? extends Marker>>> startActivation){
        this.question = question;
        this.decGraph = decGraph;
        this.syntaxGraph = syntaxGraph;
        this.nerGraph = nerGraph;
        this.roleGraph = roleGraph;
        this.nerMap = nerMap;
        this.startActivation=startActivation;

    }

}

File: semantic_decomposition_experiments/ai_challenge/graph_cache.py
from __future__ import annotations
from typing import Any, Dict, List, Optional, TYPE_CHECKING

if TYPE_CHECKING:
    from .question import Question
    from semantic_decomposition.graph.semantic_net import SemanticNet


class GraphCache:
    """
    Bundles all pre-built graph structures for one question.

    Replaces the Java GraphCache inner class; avoids repeated decomposition.
    """

    def __init__(self, question: Optional["Question"] = None) -> None:
        self.question = question
        self.dec_graph: Optional["SemanticNet"] = None
        self.syntax_graph: Optional["SemanticNet"] = None
        self.ner_graph: Optional["SemanticNet"] = None
        self.role_graph: Optional["SemanticNet"] = None
        self.ner_map: Dict[str, str] = {}
        self.start_activation: float = 1.0
